pandas/core/apply.py

```diff
@@ -396,6 +394,20 @@ def agg_list_like(self) -> FrameOrSeriesUnion:
                     "cannot combine transform and aggregation operations"
                 ) from err
             return result
+        else:
+            # Concat uses the first index to determine the final indexing order.
+            # The union of a shorter first index with the other indices causes
+            # the index sorting to be different from the order of the aggregating
+            # functions. Reindex if this is the case.
+            index_size = concatenated.index.size
+            if results[0].index.size != index_size:
+                good_index = next(
+                    result.index
+                    for result in results
+                    if result.index.size == index_size
+                )
+                concatenated = concatenated.reindex(good_index)
```

Comment:
can you not just
`reindex(good_index, copy=False)`?
which won't reindex if its already good

Reply:
I guess I had a case of premature optimization disease 🤷 Adjusted!